src/language-server/language-server.ts

```diff
@@ -70,8 +80,29 @@ export async function startLanguageServer() {
     }),
   )
 
+  // Initialize PuyaService and wait for it to be ready
+  const initializePuyaService = async (): Promise<PuyaService | undefined> => {
+    if (workspaceFolder) {
+      const workspacePath = URI.parse(workspaceFolder).fsPath
+
+      connection.console.log(`[${Constants.puyaServiceSource}]: Initializing PuyaService with workspace path: ${workspacePath}`)
+      const service = new PuyaService(workspacePath)
+
+      try {
+        await service.start()
+        connection.console.log(`[${Constants.puyaServiceSource}]: PuyaService started successfully`)
+        return service
+      } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error)
+        connection.console.error(`[${Constants.puyaServiceSource}]: Failed to start PuyaService: ${errorMessage}`)
+        return undefined
+      }
+    }
+    return undefined
+  }
+
   disposables.push(
-    connection.onInitialized(() => {
+    connection.onInitialized(async () => {
```

Comment:
```suggestion
    connection.onInitialized(() => {
```
Don't think this needs to be async?